File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/types/message/ReadMemberAttributeRequest.java
package org.opcfoundation.webserver.types.message;

import org.opcfoundation.webserver.types.UaChildId;
import org.opcfoundation.webserver.types.UaObjectId;

public class ReadMemberAttributeRequest {
    private final UaObjectId objectId;
    private final UaChildId childId;
    private final boolean isMethod;

    public ReadMemberAttributeRequest(
            UaObjectId objectId,
            UaChildId childId,
            boolean isMethod)
    {
        this.objectId = objectId;
        this.childId = childId;
        this.isMethod = isMethod;
    }

    public UaObjectId getObjectId() {
        return objectId;
    }

    public UaChildId getChildId() {
        return childId;
    }

    public boolean isMethod() {
        return isMethod;
    }
}

File: opcua-webservice-node/src/server/service/message/reactiveobject/ReadMemberAttributeRequest.ts
import { UaChildId, UaObjectId } from "../../../types";

export class ReadMemberAttributeRequest {
    private readonly _objectId: UaObjectId;
    private readonly _childId: UaChildId;
    private readonly _isMethod: boolean;

    constructor(
        objectId: UaObjectId,
        childId: UaChildId,
        isMethod: boolean) {
        this._objectId = objectId;
        this._childId = childId;
        this._isMethod = isMethod;
    }

    get objectId(): UaObjectId {
        return this._objectId;
    }

    get childId(): UaChildId {
        return this._childId;
    }

    get isMethod(): boolean {
        return this._isMethod;
    }
}